src/nvim/ops.c

```diff
@@ -2874,7 +2872,8 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
 
     /* adjust '] mark */
     curbuf->b_op_end.lnum = curwin->w_cursor.lnum - 1;
-    curbuf->b_op_end.col = bd.textcol + totlen - 1;
+    assert(totlen <= INT_MAX);
```

Comment:
I don't really like just adding a assert everytime a value is downcast, at least not where there is a more relevant _dynamic_ condition to check, i e `bd.textcol + totlen - 1 < MAXCOL` in this case  (unless this already checked above somewhere...)